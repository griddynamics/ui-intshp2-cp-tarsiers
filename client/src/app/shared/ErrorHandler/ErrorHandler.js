import React from 'react';

class ErrorHandler extends React.Component {
  state = {
    hasError: false
  };

  componentDidCatch(error, info) {
    console.log('Error Handler: ', error);
    this.setState({ hasError: true, errorMsg: info });
  }

  render() {
    const { hasError, errorMsg } = this.state;
    const { children } = this.props;

    if (hasError) {
      console.log('Error Handler: ', hasError && errorMsg);
    }
    return children;
  }
}

export default ErrorHandler;
