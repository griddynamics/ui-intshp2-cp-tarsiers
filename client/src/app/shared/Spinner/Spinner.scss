@import '../../../styles/variables';
@import '../../../styles/mixin';

.fa-sun {
  @include abs-center;

  animation: spin 4s infinite linear;
  font-size: xx-large;
  color: $color-main;
}

.spin-position {
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
}

.spinner-container {
  position: relative;
  width: 100%;
  height: 100%;
}
