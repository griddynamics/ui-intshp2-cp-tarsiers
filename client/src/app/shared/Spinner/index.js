import React from 'react';
import './Spinner.scss';

const Spinner = props => {
  const { height } = props;
  const spinnerStyle = { height };

  return (
    <div className="spinner-container" style={spinnerStyle}>
      <div className="spin-position">
        <div className="spinner">
          <i className="fas fa-sun" />
        </div>
      </div>
    </div>
  );
};

export default Spinner;
