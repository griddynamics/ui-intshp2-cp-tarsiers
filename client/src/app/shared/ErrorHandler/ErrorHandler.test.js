import React from 'react';
import { shallow } from 'enzyme';
import Spinner from '../Spinner';
import ErrorHandler from './ErrorHandler';

describe('<ErrorHandler />', () => {
  it('should match its snapshot', () => {
    const wrapper = shallow(<ErrorHandler />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should render the child component', () => {
    const wrapper = shallow(
      <ErrorHandler>
        <Spinner />
      </ErrorHandler>
    );

    expect(wrapper.find('Spinner').exists()).toBeTruthy();
  });

  it('should catch errors with componentDidCatch', () => {
    const wrapper = shallow(
      <ErrorHandler>
        <Spinner />
      </ErrorHandler>
    );
    const instance = wrapper.instance();
    const spy = jest.spyOn(instance, 'componentDidCatch');

    instance.testFunc = () => {
      throw new Error('test error message');
    };
    wrapper.find(Spinner).simulateError(new Error('test error message'));
    expect(spy).toHaveBeenCalled();
  });
});
