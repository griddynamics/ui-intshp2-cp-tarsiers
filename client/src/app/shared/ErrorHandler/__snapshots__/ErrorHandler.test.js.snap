// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrorHandler /> should match its snapshot 1`] = `""`;
